crates/rpc/rpc/src/eth/api/call.rs

```diff
@@ -87,10 +89,13 @@ where
 
         let target_block = block_number.unwrap_or(BlockId::Number(BlockNumberOrTag::Latest));
 
-        let ((cfg, block_env, _), block) =
-            futures::try_join!(self.evm_env_at(target_block), self.block_by_id(target_block))?;
+        let ((cfg, block_env, _), block) = futures::try_join!(
+            self.evm_env_at(target_block),
+            self.block_with_senders(target_block)
+        )?;
 
-        let block = block.ok_or_else(|| EthApiError::UnknownBlockNumber)?;
+        let (block, senders) =
```

Comment:
ptal at my commit on your last PR

this can be simplified a lot

`block` has a function that zips the transactions and senders

Reply:
Just saw your commit.
Changes have been made.